packages/controls/src/controls/definition.ts

```diff
@@ -74,6 +79,20 @@ export abstract class ControlDefinition<
     return data as Data
   }
 
+  resolveValue(
+    _data: DataType | undefined,
+  ): Resolvable<ResolvedValueType | undefined> {
```

Comment:
This method signature will get amended in follow-up PRs to support async resolution/side effects.